fluentd/generate_throttle_configs.rb

```diff
@@ -139,7 +143,7 @@ def close_file(project)
   tag kubernetes.*
   format json
   keep_time_key true
-  read_from_head "#{ENV['JSON_FILE_READ_FROM_HEAD'] || 'true'}"
+  read_from_head "#{ENV[READ_FROM_HEAD] || 'true'}"
```

Comment:
are you sure about this?  I don't think this is how ruby here documents/here strings work.

Reply:
what do you have in mind here? I initialize the `READ_FROM_HEAD` variable on the top of the script to contain either `JSON_FILE_READ_FROM_HEAD` or `CRIO_READ_FROM_HEAD`, therefore, read appropriate env variable.

Reply:
that isn't how ruby here doc variable interpolation is supposed to work - see https://makandracards.com/makandra/1675-using-heredoc-for-prettier-ruby-code

What does the resulting fluentd config file look like?

Reply:
I don't have my work machine with me, but trying similar code locally in ruby interpret, it works how I'd expect it to work

```ruby
irb(main):001:0> x = "PATH"
=> "PATH"
irb(main):002:0> File.open("test_file", "w") { |file|
irb(main):003:1*     file.write(<<-CONF)
irb(main):004:2"     #{ENV[x]}
irb(main):005:2"     CONF
irb(main):006:1> }
```
and when I `cat test_file` it contains the content of my `PATH` env variable

Doesn't the string interpolation in heredoc work just like string interpolation in any other ruby string? It should be pretty much just embedded piece of ruby code that gets evaluated before the string is evaluated, is it not?

Reply:
ok, you are correct